### ISSUE TITLE
A way to unzip to memory instead of fs?

### ISSUE THREAD

**Issue by AlexKovalevych** [jsonkennell/xlsxir#49]
Currently if i process multiple excel files in parallel they will be extracted to the same directory at the disk which makes it impossible to parse files. Probably we should consider extract to memory or allow to set extract directory.

**Comment by toraritte** [comment]
Changing the `:zip.extract/2` command at https://github.com/kennellroxco/xlsxir/blob/master/lib/xlsxir/unzip.ex#L149
would achieve that.

I changed `{:cwd, 'temp/'}` to `:memory` and it works in the `Unzip` module but the `temp/` directory prefix is hardcoded at multiple places in `lib/xlsxir.ex` so it will need some massaging.

```elixir
  def extract_xml_to_file(file_list, path) do
    path
    |> to_char_list
    |> :zip.extract([{:file_list, file_list}, :memory])   # {:cwd, 'temp/'}])
    |> case do
        {:error, reason}  -> {:error, reason}
        {:ok, []}         -> {:error, "Invalid File. Required XML files not found."}
        {:ok, files_list} -> {:ok, files_list}
       end
  end
```

@kennellroxco Thanks a lot for this package! The name is pretty cool as well.

**Comment by kennellroxco** [comment]
Hey @AlexKovalevych and @toraritte. I see the issue with the directory and parsing of multiple files. I originally used the `:memory` method that @toraritte pointed out, however, I found that it was possible to run into memory issues when parsing large files which is why I switched to extracting to file instead. I like the idea of making parsing to memory optional and agree that we need a solution with regard to the file save directory when parsing multiple files. I'll start working on a solution...if anyone wants to throw out some ideas or submit a pull request with some options, that would be great. Thanks!
